### PR TITLE
Cleanup test conditions and data

### PR DESCRIPTION
**Pull request by dougbu**
See individual commit descriptions for details. One-liners visible below should be enough.

I have not updated the conditions under which tests execute -- no `IsMono` -> `IsWindows` changes for example. That'll come in a later PR.

On the upside however, existing skips are better reported now.

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.Formatters.Xml.Test/XmlSerializerInputFormatterTest.cs [hunk 1]
```diff
@@ -244,15 +240,11 @@ public void HasProperSuppportedEncodings()
             await Assert.ThrowsAsync(typeof(InvalidOperationException), () => formatter.ReadAsync(context));
         }
 
-        [Fact]
+        [ConditionalFact]
+        // ReaderQuotas are not honored on Mono
+        [FrameworkSkipCondition(RuntimeFrameworks.Mono)]
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Formatters.Xml.Test/XmlSerializerInputFormatterTest.cs]: Could we set the SkipReason instead of using a comment?

Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Formatters.Xml.Test/XmlSerializerInputFormatterTest.cs]: Aww, it's not settable.

**Comment by pranavkm** [comment]
:shipit:

**Comment by dougbu** [comment]
b705922

**Pull request closed without merging**